spring-data-mongodb/src/main/java/org/springframework/data/mongodb/core/aggregation/AggregationOperationContext.java

```diff
@@ -78,6 +79,29 @@ default Document getMappedObject(Document document) {
 	 */
 	FieldReference getReference(String name);
 
+	/**
+	 * Obtain the target field name for a given field/type combination.
+	 * 
+	 * @param type The type containing the field.
+	 * @param field The property/field name
+	 * @return never {@literal null}.
+	 * @since 4.2
+	 */
+	default String getMappedFieldName(Class<?> type, String field) {
```

Comment:
Instead of adding another method that accepts a type, how about either `AggregationOperationContext.withType(…)` or reusing `getFields(Class<?> type)`?